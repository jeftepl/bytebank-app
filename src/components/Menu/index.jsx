import React from 'react'
import styles from './Menu.module.css'

const menuItems = ['Home', 'Transfers', 'Investments', 'Other services']

export default function Menu() {
	return (
		<nav className={styles.menu}>
			{menuItems.map((item, index) => {
				return (
					<div key={item} className={styles.item}>
						<a href='/' className={styles.link}>
							{item}
						</a>
						{index !== menuItems.length - 1 && <div className={styles.divider} />}
					</div>
				)
			})}
		</nav>
	)
}
