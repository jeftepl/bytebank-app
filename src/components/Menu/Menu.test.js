import { render, screen } from '@testing-library/react'
import Menu from '.'

test('Should render home page link on menu', () => {
	render(<Menu />)
	const homeLink = screen.getByText('Home')
	expect(homeLink).toBeInTheDocument()
})

test('Should render menu items', () => {
	render(<Menu />)
	const menuItems = screen.getAllByRole('link')
	expect(menuItems).toHaveLength(4)
})

test('Should not render bank statement link on menu', () => {
	render(<Menu />)
	const bankStatementLink = screen.queryByText('Statement')
	expect(bankStatementLink).not.toBeInTheDocument()
})
