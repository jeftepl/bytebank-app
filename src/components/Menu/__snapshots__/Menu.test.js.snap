// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render menu items with link classes 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Home
  </a>,
  <a
    class="link"
    href="/"
  >
    Transfers
  </a>,
  <a
    class="link"
    href="/"
  >
    Investments
  </a>,
  <a
    class="link"
    href="/"
  >
    Other services
  </a>,
]
`;
